Move ShareableResource from WebKit down to WebCore

cdumez opened the pull request:
#### 134d00115eb9067826330efabd99ecb2570f65ef
<pre>
Move ShareableResource from WebKit down to WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=268629">https://bugs.webkit.org/show_bug.cgi?id=268629</a>

Reviewed by Don Olmstead.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/ShareableResource.cpp: Renamed from Source/WebKit/Shared/ShareableResource.cpp.
(WebCore::ShareableResourceHandle::ShareableResourceHandle):
(WebCore::ShareableResource::wrapInSharedBuffer):
(WebCore::ShareableResourceHandle::tryWrapInSharedBuffer):
(WebCore::ShareableResource::create):
(WebCore::ShareableResource::map):
(WebCore::ShareableResource::ShareableResource):
(WebCore::ShareableResource::createHandle):
(WebCore::ShareableResource::data const):
(WebCore::ShareableResource::size const):
* Source/WebCore/platform/ShareableResource.h: Renamed from Source/WebKit/Shared/ShareableResource.h.
(WebCore::ShareableResourceHandle::size const):
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/ShareableResource.serialization.in: Removed.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/274025@main">https://commits.webkit.org/274025@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0f95005c49867447ade58bedb592e0a91a344166

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37556 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16441 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39819 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40082 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33454 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38874 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19063 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13604 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31843 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38121 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13851 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/32926 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12071 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/37590 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12045 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33612 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41347 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33940 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34031 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37932 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12610 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10161 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36107 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/14054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13016 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4884 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13367 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -2764,7 +2764,7 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
 #if USE(QUICK_LOOK)
     void didStartLoadForQuickLookDocumentInMainFrame(const String& fileName, const String& uti);
-    void didFinishLoadForQuickLookDocumentInMainFrame(ShareableResourceHandle&&);
+    void didFinishLoadForQuickLookDocumentInMainFrame(WebCore::ShareableResourceHandle&&);
```
Nit don't see that change in WebPageProxy.cpp.

cdumez replied:
Because it has `using namespace WebCore;`

donny-dont commented on Source/WebKit/WebProcess/Network/NetworkProcessConnection.h:
```diff
@@ -113,7 +113,7 @@ class NetworkProcessConnection : public RefCounted<NetworkProcessConnection>, IP
 #if ENABLE(SHAREABLE_RESOURCE)
     // Message handlers.
-    void didCacheResource(const WebCore::ResourceRequest&, ShareableResource::Handle&&);
+    void didCacheResource(const WebCore::ResourceRequest&, WebCore::ShareableResource::Handle&&);
```
Nit no change in .cpp

cdumez replied:
It has `using namespace WebCore;`

donny-dont commented on Source/WebKit/WebProcess/Network/WebResourceLoader.h:
```diff
@@ -100,7 +100,7 @@ class WebResourceLoader : public RefCounted<WebResourceLoader>, public IPC::Mess
     
 #if ENABLE(SHAREABLE_RESOURCE)
-    void didReceiveResource(ShareableResource::Handle&&);
+    void didReceiveResource(WebCore::ShareableResource::Handle&&);
```
Nit no change in .cpp

cdumez replied:
It has `using namespace WebCore;`

donny-dont reviewed (approved):
If bots are happy this is good to go.

Just making note of some potential breaks in a non-unified build.

cdumez committed:
Move ShareableResource from WebKit down to WebCore
https://bugs.webkit.org/show_bug.cgi?id=268629

Reviewed by Don Olmstead.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/ShareableResource.cpp: Renamed from Source/WebKit/Shared/ShareableResource.cpp.
(WebCore::ShareableResourceHandle::ShareableResourceHandle):
(WebCore::ShareableResource::wrapInSharedBuffer):
(WebCore::ShareableResourceHandle::tryWrapInSharedBuffer):
(WebCore::ShareableResource::create):
(WebCore::ShareableResource::map):
(WebCore::ShareableResource::ShareableResource):
(WebCore::ShareableResource::createHandle):
(WebCore::ShareableResource::data const):
(WebCore::ShareableResource::size const):
* Source/WebCore/platform/ShareableResource.h: Renamed from Source/WebKit/Shared/ShareableResource.h.
(WebCore::ShareableResourceHandle::size const):
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/ShareableResource.serialization.in: Removed.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/274025@main
